src/MediaPresenter.ts

```diff
@@ -0,0 +1,153 @@
+    const videoFormat = new RegExp("video/*");
+    return videoFormat.test(file.type);
+  }
+
+  @mobx.action
+  addFile(newFile: File): boolean {
+    if (MediaPresenter.isImage(newFile) || MediaPresenter.isVideo(newFile)) {
+      this.files.push(newFile);
+      this.durations.push(0);
+      this.played.push(0);
+      return true;
+    }
+    return false;
+  }
+
+  @mobx.action
+  removeFile(fileIndex: number): void {
+    if (this.files.length === 0) {
+      return;
+    }
+    let index = fileIndex % this.filesLength;
+    if (index > -1) {
+      this.files.splice(index, 1);
+      this.durations.splice(index, 1);
+      this.played.splice(index, 1);
+    }
+  }
+
+  setCustomOrder(value: boolean) {
+    this.customOrder = value;
+  }
+
+  @mobx.action
+  setDuration(fileIndex: number, duration: number): void {
+    this.durations[fileIndex % this.filesLength] = duration;
+  }
+
+  getDuration(fileIndex: number): number {
+    return this.durations[fileIndex % this.filesLength];
+  }
+
+  getFileName(fileIndex: number): string {
+    return this.files[fileIndex % this.filesLength].name;
+  }
+
+  getFile(fileIndex: number): File {
+    return this.files[fileIndex % this.filesLength];
+  }
+
+  @mobx.computed
+  get filesLength(): number {
+    return this.files.length;
+  }
+
+  @mobx.action
+  incrementFilePlayed(fileIndex: number, seconds: number): void {
+    let index = fileIndex % this.filesLength;
+    console.log("in increment " + this.played[index]);
+    console.log(this.durations[index]);
+    this.played[index] = (this.played[index] + seconds) % this.durations[index];
+  }
+
+  @mobx.action
+  resetFilePlayed(): void {
+    for (let i = 0; i < this.filesLength; i++) {
+      this.played[i] = 0;
+    }
+  }
+
+  getFilePlayed(fileIndex: number): number {
+    if (
+      fileIndex % this.filesLength === 0 ||
+      this.getDuration(fileIndex % this.filesLength) === 0
+    )
+      return 0;
+    if (
+      this.played[fileIndex % this.filesLength] %
+      this.getDuration(fileIndex % this.filesLength)
+    )
+      return 0;
+    return (
+      this.played[fileIndex % this.filesLength] %
+      this.getDuration(fileIndex % this.filesLength)
+    );
+  }
+
+  shuffleArray(): number[] {
+    for (var array = [], i = 0; i < this.files.length; ++i) array[i] = i;
+    if (this.customOrder) {
+      return array;
+    }
+    return [...array].sort(() => Math.random() - 0.5);
+  }
+  switchOrder(index: number, newIndex: any): void {
+    if (typeof newIndex === "string") {
+      newIndex = parseInt(newIndex);
+    }
+    let fileTemp: File;
+    let durTemp: number;
+    let playedTemp: number;
+    if (newIndex > -1) {
+      [fileTemp] = this.files.splice(index, 1);
+      [durTemp] = this.durations.splice(index, 1);
+      [playedTemp] = this.played.splice(index, 1);
+      this.files.splice(newIndex, 0, fileTemp);
+      this.durations.splice(newIndex, 0, durTemp);
+      this.played.splice(newIndex, 0, playedTemp);
```

Comment:
Is there a better way to handle this?

For example: what if rather than maintaining ordered arrays of information (which have to each be reordered), you create an array of MediaStore-s, where MediaStore is a class with: file, duration, isPlayed etc.. Then you would only need to worry about one index (of the MediaStore array). 
Take a look at the 'fills' object in `VideoPresenter` > `VideoStore` as an example of this in practise (there it's using a WeakMap, but for you an array would be sufficient as your key is a number)

Reply:
If you think this will require too much reworking, I'm happy to discuss

Reply:
all good I can make the videoStore inside VideoPresenter. So to clarify, you would like the videoStore to be store in a bundled form (object/map/weakmap) and then VideoPresenter has a list type of videoStore?

In that case, you are not saying that using splice is bad, but having to do it to three different variables everytime file is deleted is redundant?

Reply:
*mediaPresenter/Store, in your case

Yep exactly - any time you're doing the same thing (eg. splice) for multiple different cases, it suggests that the information is linked, and therefore there's probably a better way of handling the data all together